Source/bmalloc/libpas/src/test/PGMTests.cpp

```diff
@@ -252,6 +253,115 @@ void testPGMErrors() {
     pas_heap_lock_unlock();
 }
 
+void testPGMMetadataVectorManagement() {
+    pas_probabilistic_guard_malloc_initialize_pgm_as_enabled();
+
+    pas_heap_lock_lock();
+    pas_root* root = pas_root_create();
+    pas_heap_lock_unlock();
+
+    size_t total_allocations = 12;
+    int** int_arr = static_cast<int**>(bmalloc_allocate(total_allocations * sizeof(int*), pas_non_compact_allocation_mode));
+    // Allocate arrays of ints, of random size between [1, 30000].
+    for (size_t i = 0; i < total_allocations; ++i) {
+        int_arr[i] = static_cast<int*>(bmalloc_allocate(((rand() % 30000) + 1) * sizeof(int), pas_non_compact_allocation_mode));
+        CHECK(int_arr[i]);
+    }
+
+    pas_heap_lock_lock();
+    // Deallocate all previous allocations
+    for (size_t i = 0; i < total_allocations; ++i)
+        pas_probabilistic_guard_malloc_deallocate(int_arr[i]);
+
+    pas_ptr_hash_map* hash_map = root->pas_pgm_hash_map_instance;
+    CHECK(hash_map);
+
+    CHECK_EQUAL(hash_map->key_count, 11);
```

Comment:
If `MAX_PGM_DEALLOCATED_METADATA_ENTRIES` ever changes this test may need to be updated here.